drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -825,6 +826,41 @@ static int lsm6dsl_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int lsm6dsl_pm_action(const struct device *dev,
+			     enum pm_device_action action)
+{
+	struct lsm6dsl_data *data = dev->data;
+	int ret;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		break;
```

Comment:
shouldn't  you restore registers value same as before getting suspended?

Reply:
Sure, please take a look. I tried it with a simple polling sample and the values freeze when you suspend and continue to change when you resume it.

Personally I don't like the added ifdef, I'd rather see this function not within any ifdef. It will be optimized away if it's not used anyways. Please let me know what you think.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/73964476/c10da914-2b76-4397-92e4-e82545cb723b)